File: 백준/Silver/11047. 동전 0/동전 0.js
const fs = require('fs');
const file = process.platform === 'linux' ? './dev/stdin' : './input.txt';
const input = fs.readFileSync(file).toString().split('\n');

let [n, k] = input[0].split(' ').map(Number);
const coins = [];
let cnt = 0;

for (let i = n; i >= 1; i--) {
  coins.push(Number(input[i]));
}

for (let i = 0; i < n; i++) {
  cnt += Math.floor(k / coins[i]);
  k %= coins[i];

  if (k === 0) break;
}

console.log(cnt);

File: 백준/Silver/11047. 동전 0/동전 0.py
N, K = map(int, input().split())
coinArr = [int(input()) for _ in range(N)]
cnt = 0

for i in range(N-1, -1, -1):
    cnt += K // coinArr[i]
    K %= coinArr[i]
    if K == 0:
        break

print(cnt)
